### ISSUE TITLE
GPU Usage Details

### ISSUE THREAD

**Issue by zihaozou** [google-research/frame-interpolation#61]
Hi,
Thanks for the great work you've done. 
Currently, I am reimplementing this project using Pytorch.

When I set the batch size to 8(following the paper) for training in my Pytorch implementation, I found that my GPU usage is only 12GB. I assume the reason you set the batch size to 8 is that a batch with 8 samples can fully occupy the Nvidia V100 (16GB? or 32GB?). So, my implementation might have something wrong(e.g. lacks some layer).

I am wondering how the GPU usage is in your TensorFlow version.
I tried to run your training code to check the GPU usage by myself, but I failed to create the TFRecords file(due to storage limitations or memory overflow I think).

In my implementation, I set pyramid_levels=7, fusion_pyramid_levels=5, specialized_levels=3, flow_convs=[3, 3, 3, 3], flow_filters=[32, 64, 128, 256], sub_levels=4, filters=64, image size=256x256

**Comment by fitsumreda** [comment]
Hi, thanks for your interest in our work. The gpu usage is within my expectation. The 8 batch size is chosen based quantitative accuracy, and could have fitted a few more samples into a V100 16GB GPU.

**Comment by fitsumreda** [comment]
What errors did you run into when creating TFRecord files?

**Comment by zihaozou** [comment]
> Hi, thanks for your interest in our work. The gpu usage is within my expectation. The 8 batch size is chosen based quantitative accuracy, and could have fitted a few more samples into a V100 16GB GPU.

Thanks, then I think my implementation might be right.

> What errors did you run into when creating TFRecord files?

No error popped up. The script was suddenly killed after several minutes. I guess it is because it used too much memory(about 50Gb before it was killed) and my permission level is too low to run so much memory(I ran on my school's cluster).

One thing I also want to ask about is whether you have tried to interpolate the mid-frame at time locations other than 0.5. It doesn't look good in my implementation. I trained 200 epochs(about 160K steps) for now; the performance not at 0.5 is much worse than the performance at 0.5.

**Comment by fitsumreda** [comment]
The model is strictly trained to predict at t=0.5, to reduce the dependence on large scale high FPS training frames. If you’ve large enough high FPS data you could technically relax that requirement in training/interpolator.py.

The other alternative is to recursively invoke the model during inference, to output fractional frames.

**Comment by zihaozou** [comment]
Got it. Thanks!
